internal/controller/providers/pg/applier.go

```diff
@@ -274,6 +262,92 @@ func (p *applier) Monitoring() error {
 	return nil
 }
 
+func (p *applier) PodSchedulingPolicy() error {
+	// NOTE: this method shall be called after Engine() and Proxy() methods
+	// because it extends the engine and proxy specs with the affinity rules.
+	//
+	// The following cases are possible:
+	// 1. The user did not specify a .spec.podSchedulingPolicyName -> do nothing.
+	// 2. The user specified a .spec.podSchedulingPolicyName, but it does not exist -> return error.
+	// 3. The user specified a .spec.podSchedulingPolicyName, and it exists, but it is not applicable to the upstream cluster -> return error.
+	// 4. The user specified a .spec.podSchedulingPolicyName, and it exists, but affinity configuration is absent there -> do nothing.
+	// 5. The user specified a .spec.podSchedulingPolicyName, and it exists, and it is applicable to the upstream cluster ->
+	// copy the affinity rules to the upstream cluster spec from policy.
+
+	pg := p.PerconaPGCluster
+	// --------------------------------- //
+	// Special workaround, need to reset all affinity params in p.PerconaPGCluster before moving further.
+	// TODO: Remove it once https://perconadev.atlassian.net/browse/EVEREST-2023 is addressed
+	for i := range len(pg.Spec.InstanceSets) {
+		// all instances in sets have the same affinity configuration.
+		pg.Spec.InstanceSets[i].Affinity = nil
+	}
+	pg.Spec.Proxy.PGBouncer.Affinity = nil
+	// --------------------------------- //
+	pspName := p.DB.Spec.PodSchedulingPolicyName
+	if pspName == "" {
+		// Covers case 1.
+
+		// New affinity settings (added in 1.2.0) shall be applied only for new clusters, or on existing clusters that
+		// have been upgraded to CRVersion 2.4.1.
+		// This is a temporary workaround to make sure we can apply this change without an automatic restart.
+		// TODO: fix this once https://perconadev.atlassian.net/browse/EVEREST-1413 is addressed.
+
+		// crVersion := goversion.Must(goversion.NewVersion(pg.Spec.CRVersion))
+		// if p.DB.Status.Status == everestv1alpha1.AppStateReady &&
+		// 	crVersion.LessThan(goversion.Must(goversion.NewVersion("2.4.1"))) {
+		//
+		// 	for i := range len(pg.Spec.InstanceSets) {
+		// 		// all instances in sets have the same affinity configuration.
+		// 		pg.Spec.InstanceSets[i].Affinity = p.currentPGSpec.InstanceSets[0].Affinity
+		// 	}
+		// 	pg.Spec.Proxy.PGBouncer.Affinity = p.currentPGSpec.Proxy.PGBouncer.Affinity
+		// }
```

Comment:
Should this be uncommented?

How are we going to ensure that this new change preserves the existing affinity configuration?

Reply:
Like we discussed today, since we enforce only 2 supported operator versions per everest version and in v1.6.0 we only supported PGO v2.5.0 and v2.6.0, there can't be any PG DBs in v1.6.0 that are still using a CR version older than v2.4.1. Therefore, this is essentially dead code and can be safely removed.